reader.go

```diff
@@ -297,6 +1178,10 @@ func (r *Reader) ReadMessage(ctx context.Context) (Message, error) {
 // The function returns a lag of zero when the reader's current offset is
 // negative.
 func (r *Reader) ReadLag(ctx context.Context) (lag int64, err error) {
+	if r.config.GroupID != "" {
```

Comment:
This is tested plenty of times, how about adding a method to help make the code more readable (something like r.useConsumerGroup())